test/TestProjects/ReferenceTypes/Generated/Models/ErrorAdditionalInfo.cs

```diff
@@ -21,17 +20,17 @@ public ErrorAdditionalInfo()
         }
 
         /// <summary> Initializes a new instance of ErrorAdditionalInfo. </summary>
-        /// <param name="type"> The additional info type. </param>
+        /// <param name="errorAdditionalInfoType"> The additional info type. </param>
         /// <param name="info"> The additional info. </param>
         [SerializationConstructor]
-        internal ErrorAdditionalInfo(ResourceType type, object info)
+        internal ErrorAdditionalInfo(string errorAdditionalInfoType, object info)
         {
-            Type = type;
+            ErrorAdditionalInfoType = errorAdditionalInfoType;
             Info = info;
         }
 
         /// <summary> The additional info type. </summary>
-        public ResourceType Type { get; }
+        public string ErrorAdditionalInfoType { get; }
```

Comment:
Previously, a property with name `Type` in ReferenceType will use ResourceType type, here it's not correct. Change back to string with the changes in this PR.